src/map/lua/lua_baseentity.cpp

```diff
@@ -671,6 +671,30 @@ inline int32 CLuaBaseEntity::resetLocalVars(lua_State* L)
     return 0;
 }
 
+/************************************************************************
+*  Function: getLastOnline()
+*  Purpose : Returns the unix timestamp of the last time the char logged off or zoned
+*  Example : player:getLastOnline()
+*  Notes   :
+************************************************************************/
+
+inline int32 CLuaBaseEntity::getLastOnline(lua_State* L)
+{
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity == nullptr);
+    TPZ_DEBUG_BREAK_IF(m_PBaseEntity->objtype != TYPE_PC);
+
+    if (auto PChar = static_cast<CCharEntity*>(m_PBaseEntity); PChar->objtype == TYPE_PC)
```

Comment:
`static_cast` forces the cast without any checking, so if debug breaks are off and a mob is passed in here, this check will still pass. Granted, it'll get caught by `PChar->objtype == TYPE_PC`, but it's a practice we shouldn't be relying on;

```cpp
if (auto* PChar = dynamic_cast<CCharEntity*>(m_PBaseEntity))
{
    ....
}
```

will honour nullptr, and check that the pointer is in the same inheritance tree, rather than hard-casting the pointer by virtue of them being the same size.